### PR TITLE
Eventviewer generic

### PR DESCRIPTION
**Pull request by jeffj6123**
This will provide a "second view" where every event in the  event store is reflected into the timeline by taking the timestamp and with addition of a few optional properties can be formatted.

event store all

Events are broken down into Event Category/ Type( pairs

The optional properties as of right now are
* Status : Color the dot according to 3 potential values. "Ok" = Green, "Warning" = "Yellow", "Error" = Red. Omitting this value will default to a white circle. Additionally health report based events are checked according to their HealthState value.
* Duration : A number value that will be subtracted from the current time stamp and create a bar instead of a point in time.
* Description : Coupled with duration will be the text displayed over the bar instead of being blank.
some structured data
 i.e
        {
            "Kind": "ClusterUpgradeDomainCompleted",
            "TargetClusterVersion": "6.5.676.9590:31",
            "UpgradeState": "RollingForward",
            "UpgradeDomains": "(4)",
            "UpgradeDomainElapsedTimeInMs": 60186.2683,
            "EventInstanceId": "0bd73fa6-9d97-4987-b497-3d5f7de2f590",
            "TimeStamp": "2019-11-11T22:30:59.8005863Z",
            "Category": "Upgrade",
            "HasCorrelatedEvents": false,
            **"Description": "some data",
            "Status": "Error",
            "Duration": 100000**
        }

### CODE REVIEW

File: src/Sfx-Proxy/appsettings.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "TargetCluster": {
-    "Url": "http://localhost:19080"
+  "Url": "https://jejarryacesstesting.eastus.cloudapp.azure.com:19080",
```
Comment (Christina-Kang) [inline src/Sfx-Proxy/appsettings.json]: left in by mistake?

File: src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+    Reference for understanding the timeline rendering library https://visjs.github.io/vis-timeline/docs/timeline/
+
+    high level concerns:
+
+    Currently there are 2 "versions" of the timeline event generation.
+    1. Applies some level of parsing event/multiple events to create one entry on the timeline. This is currently only used on significant items
+    2. Generically applies all events to the timeline. Some level of customization is available to drive their appearance from information on the event. Currently by adding "Duration", "Status", "Description" properties.
+
+    Ideally enough flexibility is applied to the generic handling to have all logic driven from there and no needing to update SFX except for updating to look for more fields.
+
```
Comment (Christina-Kang) [inline src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: nit: remove extra line

File: src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts [hunk 3]
```diff
@@ -116,13 +136,38 @@ module Sfx {
...
                 type: "range",
-                title: EventStoreUtils.tooltipFormat(event, start, end),
+                title: EventStoreUtils.tooltipFormat(event.eventProperties, start, end),
                 className: rollBack  ? "orange" : "green"
             });
         }
+
+        /*
+        add the subgroup stacking to every group so now we can always reliably place stacking/nonstacking items in any group.
+
```
Comment (Christina-Kang) [inline src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: nit: remove extra line

File: src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts [hunk 4]
```diff
@@ -382,4 +426,147 @@ module Sfx {
...
+            }else{
+                i = properties.length;
+                constructedPath = ""
+            }
+        }
+
+        return constructedPath
+    }
+
+
```
Comment (Christina-Kang) [inline src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: nit: remove extra line

File: src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts [hunk 5]
```diff
@@ -382,4 +426,147 @@ module Sfx {
...
+                    item.type = "range";
+                    item.className = color;
+                    item.subgroup = "stack";
+                }catch (e ) {}
+            }
+
+            items.add(item);
+        });
+
+        console.log(groupIds)
```
Comment (Christina-Kang) [inline src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: intentional

Comment (jeffj6123) [reply src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: it is not.

**Pull request merged**